modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -58,6 +58,13 @@ final class AzureStorageSettings {
         key -> SecureSetting.secureString(key, null)
     );
 
+    /** Azure instance discovery */
+    public static final AffixSetting<Boolean> INSTANCE_DISCOVERY_ENABLED_SETTING = Setting.affixKeySetting(
+        AZURE_CLIENT_PREFIX_KEY,
+        "instance_discovery.enabled",
+        key -> Setting.boolSetting(key, true, Property.NodeScope)
+    );
```

Comment:
I think we should probably have this as a sysprop rather than a setting, it only really makes sense in tests.

Reply:
I'm not positive that it's only good for tests. Depending on the setup our customers have, they might not want instance discovery:

https://learn.microsoft.com/en-us/java/api/com.azure.identity.defaultazurecredentialbuilder?view=azure-java-stable#com-azure-identity-defaultazurecredentialbuilder-disableinstancediscovery()

> Disables the setting which determines whether or not instance discovery is performed when attempting to authenticate. This will completely disable both instance discovery and authority validation. This functionality is intended for use in scenarios where the metadata endpoint cannot be reached, such as in private clouds or Azure Stack. The process of instance discovery entails retrieving authority metadata from https://login.microsoft.com/ to validate the authority. By utilizing this API, the validation of the authority is disabled. As a result, it is crucial to ensure that the configured authority host is valid and trustworthy.

That said, I don't have much context here so I'm good with turning this into a system prop instead. Let me know if given the above description, we still don't want this as a setting.

Reply:
Converted to a system prop -- we can always promote this to a setting later, if need be.